lib/nexpose/ajax.rb

```diff
@@ -158,14 +158,28 @@ def request(nsc, request, timeout = nil)
           raise Nexpose::AuthenticationFailed.new(response)
         else
           req_type = request.class.name.split('::').last.upcase
-          raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{request.body}", response.code)
+          raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{get_reason(request, response)}", response.code)
         end
       else
         req_type = request.class.name.split('::').last.upcase
-        raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{request.body}", response.code)
+        raise Nexpose::APIError.new(response, "#{req_type} request to #{request.path} failed. #{get_reason(request, response)}", response.code)
       end
     end
 
+    def get_api_version(request)
+      begin
+        matches = request.path.match(/\/api\/(?<version>[\d\.]+)\//)
+        matches[:version].to_f
+      rescue
+        0.0
+      end
+    end
+
+    def get_reason(request, response)
+      version = get_api_version(request)
+      (version >= 2.1 && response.body) ? response.body : request.body
```

Comment:
How about decorating the reason instead of just adding the response/request body? Something to show which is being printed?